### PR TITLE
Upgrade v1.1.1

### PR DESCRIPTION
**Pull request by KarishmaBothara**

### CODE REVIEW

File: packages/api/test/e2e/api.calls.e2e.ts [hunk 1]
```diff
@@ -64,8 +64,6 @@ describe('e2e api calls', () => {
       expect(currentSession.eraLength.toNumber()).toBeGreaterThanOrEqual(0);
       expect(currentSession.eraProgress.toNumber()).toBeGreaterThanOrEqual(0);
       expect(currentSession.isEpoch).toBe(true);
-      expect(currentSession.lastEraLengthChange.toNumber()).toBeGreaterThanOrEqual(0);
```
Comment (KarishmaBothara) [inline packages/api/test/e2e/api.calls.e2e.ts]: support for last era length change and last length change is removed from polkadots

File: packages/types/src/extrinsic/types.ts [hunk 2]
```diff
@@ -75,14 +74,16 @@ export interface ExtrinsicPayloadValue {
 
 export type DoughnutValue = AnyU8a;
 
+// export type ChargeTransactionValue = AnyU8a;
```
Comment (holygits) [inline packages/types/src/extrinsic/types.ts]: Remove?

File: packages/types/src/extrinsic/types.ts [hunk 3]
```diff
@@ -75,14 +74,16 @@ export interface ExtrinsicPayloadValue {
...
+
 export type FeeExchangeValue = {
   assetId: AnyNumber;
   maxPayment: AnyNumber;
 };
 
 export interface IExtrinsicImpl extends IExtrinsicImplBase {
```
Comment (holygits) [inline packages/types/src/extrinsic/types.ts]: Can remove this interface altogether

File: packages/types/src/extrinsic/v2/Extrinsic.ts [hunk 4]
```diff
@@ -14,57 +14,73 @@
...
     );
   }
 
-  static decodeExtrinsic(value?: Call | Uint8Array | ExtrinsicValueV2, isSigned: boolean = false): ExtrinsicValueV2 {
+  static decodeExtrinsic(
+    registry: Registry,
+    value?: Call | Uint8Array | ExtrinsicValueV2,
+    isSigned: boolean = false
+  ): ExtrinsicValueV2 {
+    // const registry = new TypeRegistry();
```
Comment (holygits) [inline packages/types/src/extrinsic/v2/Extrinsic.ts]: remove

File: packages/types/src/extrinsic/v2/Extrinsic.ts [hunk 5]
```diff
@@ -14,57 +14,73 @@
...
       return {method: value as Call};
     } else if (isU8a(value)) {
       // here we decode manually since we need to pull through the version information
-      let v = value;
-      const signature = new ExtrinsicSignatureV2(v, {isSigned});
-      v = v.subarray(signature.encodedLength);
-      const method = createType('Call', v);
+      // let v = value;
+      const signature = new ExtrinsicSignatureV2(registry, value, {isSigned});
+      // v = v.subarray(signature.encodedLength);
```
Comment (holygits) [inline packages/types/src/extrinsic/v2/Extrinsic.ts]: Remove commented stuff

File: packages/types/src/extrinsic/v2/Extrinsic.ts [hunk 6]
```diff
@@ -116,4 +140,9 @@ export default class ExtrinsicV2 extends Struct implements IExtrinsicImpl {
 
     return this;
   }
+
+  toU8a(isBare?: boolean): Uint8Array {
```
Comment (holygits) [inline packages/types/src/extrinsic/v2/Extrinsic.ts]: can remove this method

File: packages/types/src/extrinsic/v2/ExtrinsicSignature.ts [hunk 7]
```diff
@@ -144,47 +164,113 @@ export default class ExtrinsicSignatureV2 extends Struct implements IExtrinsicSi
     signature: Uint8Array | string,
     payload: ExtrinsicPayloadValueV2 | Uint8Array | string
   ): IExtrinsicSignature {
+    //console.log('this.registry:',this.registry);
```
Comment (holygits) [inline packages/types/src/extrinsic/v2/ExtrinsicSignature.ts]: Remove

File: packages/types/src/extrinsic/v2/ExtrinsicSignature.ts [hunk 8]
```diff
@@ -144,47 +164,113 @@ export default class ExtrinsicSignatureV2 extends Struct implements IExtrinsicSi
...
+      tip,
+      transactionPayment,
+    } as ExtrinsicV2SignatureOptions);
+    const signature = createType(
+      this.registry,
+      'MultiSignature',
+      u8aConcat(new Uint8Array([1]), new Uint8Array(64).fill(0x42))
+    );
+
+    // const payload = new ExtrinsicPayloadV2(this.registry, payloadValue);
```
Comment (holygits) [inline packages/types/src/extrinsic/v2/ExtrinsicSignature.ts]: Remove

File: packages/types/src/injects.ts [hunk 9]
```diff
@@ -11,30 +11,28 @@
...
 // See the License for the specific language governing permissions and
 // limitations under the License.
-import {u64} from '@polkadot/types';
+import {TypeRegistry, u64} from '@polkadot/types';
 import Option from '@polkadot/types/codec/Option';
+import {MultiSignature} from '@polkadot/types/interfaces/runtime/types';
 import Doughnut from './Doughnut';
 import * as extrinsicTypes from './extrinsic';
 import * as runtimeTypes from './runtime';
+//import { ChargeTransactionPayment } from './runtime';
```
Comment (holygits) [inline packages/types/src/injects.ts]: remove

File: packages/types/src/injects.ts [hunk 10]
```diff
@@ -11,30 +11,28 @@
...
-  u8a.set([1]);
-  u8a.set(this.raw.toU8a(true), 1);
-  return u8a;
-};
 
 export default {
   ...runtimeTypes,
   ...extrinsicTypes,
-  AssetOf: 'u128',
+  // AssetOf: 'u128',
```
Comment (holygits) [inline packages/types/src/injects.ts]: Remove commented stuff

File: packages/api/src/derives/index.ts [hunk 11]
```diff
@@ -19,23 +19,21 @@ import * as genericAsset from '@cennznet/crml-generic-asset/derives';
 import {AnyFunction} from '@cennznet/types/types';
 import {ApiInterfaceRx, MethodResult} from '@polkadot/api/types';
 import {Observable} from 'rxjs';
-import * as fees from './fees';
 
 export type DeriveFunc = (api: ApiInterfaceRx) => (...args: any[]) => Observable<any>;
 
-export const derive = {attestation, cennzxSpot, fees, genericAsset};
+// export const derive = {attestation, cennzxSpot, fees, genericAsset};
```
[the remark this answers is not in the document]
Comment (KarishmaBothara) [reply packages/api/src/derives/index.ts]: had removed this earlier..

File: jest/initApiPromise.js [hunk 12]
```diff
@@ -1,15 +1,21 @@
 import {WsProvider} from '@polkadot/api';
 import config from '../config';
 import {Api as ApiPromise} from '../packages/api/src/Api';
+import ExtrinsicSignatureV2 from '../packages/types/src/extrinsic/v2/ExtrinsicSignature';
```
Comment (holygits) [inline jest/initApiPromise.js]: Is this import needed?

File: packages/api/test/e2e/tx.e2e.ts [hunk 13]
```diff
@@ -166,13 +147,10 @@ describe('e2e transactions', () => {
...
-          .transfer(16001, receiver.address, 100)
-          .signAndSend(senderKeypair, {transactionPayment}, ({events, status}) => {
-            console.log('Transaction status:', status.type);
+          .transfer(16001, bob.address, 100)
+          .signAndSend(alice, {transactionPayment}, ({events, status}) => {
             if (status.isFinalized) {
-              console.log('Completed at block hash', status.value.toHex());
-              console.log('Events:');
-
+              // console.log('Completed at block hash', status.value.toHex());
```
Comment (holygits) [inline packages/api/test/e2e/tx.e2e.ts]: Uncomment or remove

File: packages/types/src/extrinsic/Extrinsic.ts [hunk 14]
```diff
@@ -263,7 +277,7 @@ export default class Extrinsic extends Base<ExtrinsicImpl> implements IExtrinsic
    * @param isBare true when the value has none of the type-specific prefixes (internal)
    */
   toU8a(isBare?: boolean): Uint8Array {
-    const encoded = u8aConcat(new Uint8Array([this.version]), this.raw.toU8a(isBare));
+    const encoded = u8aConcat(new Uint8Array([this.version]), this.raw.toU8a());
```
Comment (holygits) [inline packages/types/src/extrinsic/Extrinsic.ts]: Is this change required?

File: packages/types/src/runtime/cennzx/cennzx.spec.ts [hunk 15]
```diff
@@ -12,25 +12,27 @@
...
+import {createTypeUnsafe, TypeRegistry} from '@polkadot/types';
 
 import * as cennzxTypes from './';
 import ExchangeKey from './ExchangeKey';
 import FeeRate from './FeeRate';
 import {IExchangeKey} from './types';
 
-const registry = getTypeRegistry();
+const registry = new TypeRegistry();
+// registry.register(typeof ExchangeKey, ExchangeKey);
```
Comment (holygits) [inline packages/types/src/runtime/cennzx/cennzx.spec.ts]: Remove comments

File: packages/api/src/Api.ts [hunk 16]
```diff
@@ -23,113 +23,110 @@ import derives from './derives';
...
-import {decorateExtrinsics} from './util/customDecorators';
+// import {decorateExtrinsics} from './util/customDecorators';
 import {mergeDeriveOptions} from './util/derives';
 import {getProvider} from './util/getProvider';
 import {getTimeout} from './util/getTimeout';
 import {injectOption, injectPlugins, mergePlugins} from './util/injectPlugin';
 import logger from './util/logging';
 
-export const DEFAULT_TIMEOUT = 10000;
+export const DEFAULT_TIMEOUT = 100000;
```
[the remark this answers is not in the document]
Comment (KarishmaBothara) [reply packages/api/src/Api.ts]: Changed

File: packages/types/src/extrinsic/v2/ExtrinsicSignature.ts (hunk 8 above)
Comment (KarishmaBothara) [reply packages/types/src/extrinsic/v2/ExtrinsicSignature.ts]: done

File: jest/initApiPromise.js (hunk 12 above)
Comment (KarishmaBothara) [reply jest/initApiPromise.js]: no more needed.

File: packages/types/src/extrinsic/Extrinsic.ts (hunk 14 above)
Comment (KarishmaBothara) [reply packages/types/src/extrinsic/Extrinsic.ts]: removed the mtd toU8a

File: packages/types/src/runtime/cennzx/cennzx.spec.ts (hunk 15 above)
Comment (KarishmaBothara) [reply packages/types/src/runtime/cennzx/cennzx.spec.ts]: done

File: packages/api/test/e2e/tx.e2e.ts [hunk 17]
```diff
@@ -17,39 +17,23 @@
 // of the Apache-2.0 license. See the LICENSE file for details.
 
 import {AssetId, AssetOptions, FeeExchangeV1} from '@cennznet/types';
-import {SimpleKeyring, Wallet} from '@cennznet/wallet';
-import {SubmittableResult} from '@polkadot/api';
+import ExtrinsicSignatureV2 from '@cennznet/types/extrinsic/v2/ExtrinsicSignature'
```
[the remark this answers is not in the document]
Comment (KarishmaBothara) [reply packages/api/test/e2e/tx.e2e.ts]: removed

File: packages/crml-generic-asset/test/ga.e2e.ts [hunk 18]
```diff
@@ -16,42 +16,43 @@
...
-        api.setSigner(wallet);
-        ga = api.genericAsset;
+      await cryptoWaitReady();
+      const keyring = new Keyring({ type: 'sr25519' });
+      alice = keyring.addFromUri('//Alice');
+      bob = keyring.addFromUri('//Bob');
+      api = await Api.create(
+        {provider: url,
+          types: {
+            ExtrinsicSignatureV4: ExtrinsicSignatureV2,
```
[the remark this answers is not in the document]
Comment (KarishmaBothara) [reply packages/crml-generic-asset/test/ga.e2e.ts]: yes..removed

File: packages/api/test/e2e/tx.e2e.ts (hunk 13 above)
Comment (KarishmaBothara) [reply packages/api/test/e2e/tx.e2e.ts]: removed

**Comment by KarishmaBothara** [comment]
> Still a few things to clean up

done
